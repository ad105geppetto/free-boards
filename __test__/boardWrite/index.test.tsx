import "@testing-library/jest-dom";
import { act, render } from "@testing-library/react";
import BoardWriteUI from "../../src/components/units/board/write/BoardWrite.presenter";
import { propsWriteMock, propsEditMock } from "./index.mock";

jest.mock("next/router", () => ({ useRouter: jest.fn() }));

describe("자유게시판 등록 페이지 Unit Test", () => {
  it("게시판 등록 페이지 스냅샷 테스트", async () => {
    await act(async () => {
      const result = render(<BoardWriteUI {...propsWriteMock} />);
      expect(result.container).toMatchSnapshot();
    });
  });
});

describe("자유게시판 수정 페이지 Unit Test", () => {
  it("게시판 수정 페이지 스냅샷 테스트", async () => {
    await act(async () => {
      const result = render(<BoardWriteUI {...propsEditMock} />);
      expect(result.container).toMatchSnapshot();
    });
  });
});
