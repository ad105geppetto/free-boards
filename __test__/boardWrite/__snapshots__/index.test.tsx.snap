// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`자유게시판 등록 페이지 Unit Test 게시판 등록 페이지 스냅샷 테스트 1`] = `
<div>
  <div
    class="css-1ol1jm3"
  >
    <section
      class="css-13c9wjq"
    >
      <h3
        class="css-lt8ptt"
      >
        게시글 
        등록
      </h3>
      <div
        class="css-1d57vsu"
      >
        <input
          aria-label="작성자명"
          class="css-4uhr0z"
          placeholder="작성자명을 입력해주세요."
          type="text"
          value=""
        />
        <input
          aria-label="비밀번호"
          autocomplete="off"
          class="css-4uhr0z"
          placeholder="비밀번호를 입력해주세요."
          type="password"
        />
      </div>
      <input
        aria-label="제목"
        class="css-4uhr0z"
        placeholder="제목을 입력해주세요."
        type="text"
        value=""
      />
      <input
        aria-label="내용"
        class="css-vbucm9"
        placeholder="내용을 입력해주세요."
        type="text"
        value=""
      />
      <a
        class="css-964atj"
        href="/free-boards/new"
      >
        등록하기
      </a>
    </section>
  </div>
</div>
`;

exports[`자유게시판 수정 페이지 Unit Test 게시판 수정 페이지 스냅샷 테스트 1`] = `
<div>
  <div
    class="css-1ol1jm3"
  >
    <section
      class="css-13c9wjq"
    >
      <h3
        class="css-lt8ptt"
      >
        게시글 
        수정
      </h3>
      
      <input
        aria-label="제목"
        class="css-4uhr0z"
        placeholder="제목을 입력해주세요."
        type="text"
        value="게시글 제목"
      />
      <input
        aria-label="내용"
        class="css-vbucm9"
        placeholder="내용을 입력해주세요."
        type="text"
        value="게시글 내용"
      />
      <a
        class="css-964atj"
        href="/free-boards/undefined/edit"
      >
        수정하기
      </a>
    </section>
  </div>
</div>
`;
